src/main/java/com/yourorganization/maven_sample/MyAnalysis.java

```diff
@@ -1,28 +1,41 @@
 package com.yourorganization.maven_sample;
 
+import com.github.javaparser.JavaParser;
+import com.github.javaparser.ParseStart;
+import com.github.javaparser.ParserConfiguration;
+import com.github.javaparser.StringProvider;
 import com.github.javaparser.ast.CompilationUnit;
 import com.github.javaparser.ast.expr.Expression;
 import com.github.javaparser.ast.stmt.ReturnStmt;
-import com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade;
-import com.github.javaparser.symbolsolver.model.typesystem.Type;
+import com.github.javaparser.resolution.types.ResolvedType;
+import com.github.javaparser.symbolsolver.JavaSymbolSolver;
 import com.github.javaparser.symbolsolver.resolution.typesolvers.CombinedTypeSolver;
 import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
 
 import java.io.IOException;
 
-import static com.github.javaparser.JavaParser.parse;
-
 /**
  * Some code that uses JavaSymbolSolver.
  */
 public class MyAnalysis {
-    public static void main(String[] args) throws IOException {
+
+    private static JavaParser settingUpSymbolResolution() {
         // Set up a minimal type solver that only looks at the classes used to run this sample.
         CombinedTypeSolver combinedTypeSolver = new CombinedTypeSolver();
         combinedTypeSolver.add(new ReflectionTypeSolver());
 
+        // Configure JavaParser to use type resolution
+        ParserConfiguration parserConfiguration = new ParserConfiguration();
+        parserConfiguration.setSymbolResolver(new JavaSymbolSolver(combinedTypeSolver));
+        return new JavaParser(parserConfiguration);
+    }
+
+    public static void main(String[] args) throws IOException {
+        JavaParser javaParser = settingUpSymbolResolution();
+
         // Parse some code
-        CompilationUnit cu = parse("class X { int x() { return 1 + 1.0; } }");
+        CompilationUnit cu = javaParser.parse(ParseStart.COMPILATION_UNIT,
+                new StringProvider("class X { int x() { return 1 + 1.0; } }")).getResult().get();
```

Comment:
Is this change necessary? I think we have JavaParser.setConfiguration (or use JavaParser.getConfiguration and send it to the setup method.)